synapse/lib/cell.py

```diff
@@ -4053,3 +4069,258 @@ async def _runReloadFunc(self, name, func):
+        if valid is False:
+            return False, {'mesg': f'API Key shadow mismatch: {iden}',
+                           'user': user, 'name': udef.get('name')}
+
+        return True, {'kdef': kdef, 'udef': udef}
+
+    async def modUserApiKey(self, iden, key, valu):
+        '''
+        Update a value in the user API key metadata.
+
+        Args:
+            iden (str): Iden of the key to update.
+            key (str): Name of the valu to update.
+            valu: The new value.
+
+        Returns:
+            dict: An updated key metadata dictionary.
+        '''
+        if key not in ('name',):
+            raise s_exc.BadArg(mesg=f'Cannot set {key} on user API Keys.', name=key)
+
+        kdef = await self.getUserApiKey(iden)
+        if kdef is None:
+            raise s_exc.NoSuchIden(mesg=f'User API Key does not exist, cannot modify it: {iden}', iden=iden)
+        useriden = kdef.get('user')
+        user = await self.auth.reqUser(useriden)
+
+        vals = {
+            'updated': s_common.now()
+        }
+        if key == 'name':
+            vals['name'] = valu
+
+        kdef.update(vals)
+        kdef = s_schemas.reqValidUserApiKeyDef(kdef)
+
+        await self._push('user:apikey:edit', kdef.get('user'), iden, vals)
+
+        logger.info(f'Updated HTTP API Key {iden} for {user.name}, set {key}={valu}',
+                    extra=await self.getLogExtra(target_user=user.iden, target_username=user.name, iden=iden,
+                                                 status='MODIFY'))
+
+        kdef.pop('shadow')
+        return kdef
+
+    @s_nexus.Pusher.onPush('user:apikey:edit')
+    async def _setUserApiKey(self, user, iden, vals):
+        lkey = s_common.uhex(user) + b'apikey' + s_common.uhex(iden)
+        buf = self.slab.get(lkey, db=self.usermetadb)
+        if buf is None:  # pragma: no cover
+            raise s_exc.NoSuchIden(mesg=f'User API Key does not exist: {iden}')
+        kdef = s_msgpack.un(buf)
+        kdef.update(vals)
+        self.slab.put(lkey, s_msgpack.en(kdef), db=self.usermetadb)
+        # TODO Timecache clear
+        return kdef
+
+    async def delUserApiKey(self, iden):
+        '''
+        Delete an existing API key.
+
+        Args:
+            iden (str): The iden of the API key to delete.
+
+        Returns:
+            bool: True indicating the key was deleted.
+        '''
+        kdef = await self.getUserApiKey(iden)
+        if kdef is None:
+            raise s_exc.NoSuchIden(mesg=f'User API Key does not exist: {iden}')
+        useriden = kdef.get('user')
+        user = await self.auth.reqUser(useriden)
+        ret = await self._push('user:apikey:del', useriden, iden)
+        logger.info(f'Deleted HTTP API Key {iden} for {user.name}',
+                    extra=await self.getLogExtra(target_user=user.iden, target_username=user.name, iden=iden,
+                                                 status='MODIFY'))
```

Comment:
Should this be status='DELETE'?

Reply:
I suppose it is modifying the username so MODIFY makes sense

Reply:
Specifically its modifying a user controlled resource.